decorator.rb

```diff
@@ -0,0 +1,15 @@
+# decorator.rb
+require_relative 'nameable'
+
+class Decorator < Nameable
+  attr_reader :nameable
```

Comment:
- [ ] Hello, fabulous job on implementing the decorator class! 🎉 In your current implementation, you've provided only Getter methods for the @nameable instance. However, it's important to note that in this specific instance, we require both read and write access because we'll be modifying the @nameable instance. To enhance the functionality, it would be more beneficial to have both Getters and Setters for @nameable. This way, we can not only retrieve information but also update it when necessary, making our code more versatile and adaptable. 🛠️🔄

```suggestion
 attr_accessor :nameable
```

Reply:
Hello! 🎉 Thank you for your thoughtful review and the positive feedback on implementing the decorator class! I've made the suggested change by adding both getters and setters for the @nameable instance.